pudb/var_view.py

```diff
@@ -432,7 +432,7 @@ def walk_value(self, parent, label, value, id_path=None, attr_prefix=None):
                     else:
                         key_it = xrange(len_value)
 
-            if key_it is not None:
+            if key_it is not None and hasattr(value, '__getitem__'):
```

Comment:
Wouldn't it be more robust/Pythonic to handle failure of `value[key]`?

Reply:
I'll take a look at this. I've also encountered some other similar instances where we could be a bit more robust about catching errors when trying to figure out how to walk a value.